components/TextFields/src/MDCInputTextField.h

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+/**
+ A set of Contained Input View states outlined in the Material guidelines. These states overlap with
+ and extend UIControlState.
+ */
+typedef NS_OPTIONS(NSInteger, MDCContainedInputViewState) {
```

Comment:
> const UIControlState UIControlStateError = (1 << 22);

I'm supportive of this being the direction we take in service to minimizing the amount of new API surface area we create here.

Reply:
@romoore pointed out that activated just means that there's text and that sorta blew my mind. In light of this, I'm beginning to think that it can go away as a distinct state. And now that iOS 9 is our minimum OS we can easily use `UIControlStateFocused`. If we want to just add `UIControlStateError` I might just be on board with that! Let me think about it for a bit.

Reply:
If Rob's assessment that activated just means `.text != nil` is correct, can we assume that that's the same for every component? Does it every mean anything other than that? Does anyone know?

Reply:
I do know, and it's not the case for every component. For example Tabs supports both "Activated" and "Selected". These depend on the effect of the Tabs (e.g., does it change "navigation"), but since Tabs has no text input it cannot depend on the input text to the component.

Reply:
The last time I checked, UIControlStateFocused was not used for an element that has first responder status.  In fact, I don't believe UIControlStateFocused is used on iOS at all (that I can determine.  Here's a UITextField when it is currently editing with text being input:

```

(lldb) p textField.isEditing
(BOOL) $2 = YES
(lldb) p textField.isFocused
(BOOL) $3 = NO
(lldb) 
```

Reply:
So maybe we should stick to what I originally proposed--a way of representing states that is specific to contained input views--and we should just get rid of activated

Reply:
That's a good point. Material meaning of "Focused" is not the same as the iOS implementation, making this UIControlState unusable for Material theming. 
We may want to fallback to the iOS native behavior and call this state "Editing" with API like "isEditing" and "UIControlStateEditing".  That's more intuitive to iOS users (A similar approach to us using the native "highlighted" state instead of the Material "pressed" state).

Reply:
Because it is a subclass of UITextField, it has isEditing. UIControlStateEditing seems unnecessary to me.

Reply:
Yes, I also don't see an `UIControlStateEditing` in the headers.